src/bitcoin-cli.cpp

```diff
@@ -291,6 +293,146 @@ class GetinfoRequestHandler: public BaseRequestHandler
+        // Count peer connection totals, and if m_verbose is true, store peer data in a vector of structs.
+        std::vector<m_peer> peers;
+        int ipv4_i{0}, ipv6_i{0}, onion_i{0}, block_relay_i{0}, total_i{0}; // inbound conn counters
+        int ipv4_o{0}, ipv6_o{0}, onion_o{0}, block_relay_o{0}, total_o{0}; // outbound conn counters
+        const UniValue& getpeerinfo{batch[ID_PEERINFO]["result"]};
+        for (const UniValue& peer : getpeerinfo.getValues()) {
+            const std::string addr{peer["addr"].get_str()};
+            const std::string addr_local{peer["addrlocal"].isNull() ? "" : peer["addrlocal"].get_str()};
+            const int mapped_as{peer["mapped_as"].isNull() ? 0 : peer["mapped_as"].get_int()};
+            const bool is_block_relay{!peer["relaytxes"].get_bool()};
+            const bool is_inbound{peer["inbound"].get_bool()};
+            m_conn_type conn_type{m_conn_type::ipv4};
+            if (is_inbound) {
+                if (IsAddrIPv6(addr)) {
+                    conn_type = m_conn_type::ipv6;
+                    ipv6_i += 1;
+                } else if (IsInboundOnion(mapped_as, addr, addr_local)) {
+                    conn_type = m_conn_type::onion;
+                    onion_i += 1;
+                } else {
+                    ipv4_i += 1;
+                }
+                if (is_block_relay) block_relay_i += 1;
+            } else {
+                if (IsAddrIPv6(addr)) {
+                    conn_type = m_conn_type::ipv6;
+                    ipv6_o += 1;
+                } else if (IsOutboundOnion(addr)) {
+                    conn_type = m_conn_type::onion;
+                    onion_o += 1;
+                } else {
+                    ipv4_o += 1;
+                }
+                if (is_block_relay) block_relay_o += 1;
+            }
+            if (m_verbose) {
+                const int peer_id{peer["id"].get_int()};
+                const double min_ping{peer["minping"].isNull() ? 0 : peer["minping"].get_real()};
+                const double ping{peer["pingtime"].isNull() ? 0 : peer["pingtime"].get_real()};
+                peers.push_back({peer_id, mapped_as, min_ping, ping, addr, conn_type, !is_inbound, is_block_relay});
+            }
+        }
+        // Generate reports
+        std::string result;
+        // Report 1: peer connections sorted by direction and minimum ping time
+        if (m_verbose) {
+            result += "Peer connections sorted by direction and min ping\n  id  <->  relay   conn  min ping    ping     asmap  address\n";
+            std::sort(peers.begin(), peers.end());
+            for (const m_peer& peer : peers) {
+                result += strprintf("%4i  %3s  %5s  %5s  %-8d  %-8d  %6s  %s\n", peer.id, peer.is_outbound ? "out" : "in", peer.is_block_relay ? "block" : "full", ConnTypeEnumToString(peer.conn_type), peer.min_ping, peer.ping, peer.mapped_as == 0 ? "" : ToString(peer.mapped_as), peer.addr);
```

Comment:
Probably needs a unit behind the `min ping` and `ping` values.

Reply:
done by adding units in a line at the bottom to save on horizontal space